platforms/common/spi.cpp

```diff
@@ -87,7 +87,7 @@ const px4_spi_bus_t *px4_spi_buses{nullptr};
 
 int px4_find_spi_bus(uint32_t devid)
 {
-	for (int i = 0; ((px4_spi_bus_t *) px4_spi_buses) != nullptr && i < SPI_BUS_MAX_BUS_ITEMS; ++i) {
+	for (int i = 0; i < SPI_BUS_MAX_BUS_ITEMS; ++i) {
```

Comment:
This can be nullptr if `BOARD_NUM_SPI_CFG_HW_VERSIONS > 1` and the method is called early.
Can you just ignore the error?